src/components/stack-chart/Canvas.js

```diff
@@ -108,7 +108,7 @@ class StackChartCanvasImpl extends React.PureComponent<Props> {
   // When the user checks the "use same widths for each stack" checkbox, some
   // expensive computation happens when the canvas is drawn. These computations
   // can be reused for hit testing, and therefore are saved in these variables.
-  _sameWidthsIndexAtStart: null | number;
+  _sameWidthsIndexAtViewportStart: null | number;
```

Comment:
The last commit is difficult to understand  a bit. Can you maybe add a comment here why you needed the index at viewport start?